components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -288,10 +290,13 @@ void RewardsNotificationServiceImpl::OnGrant(RewardsService* rewards_service,
                                              Grant properties) {
 if (error_code == ledger::Result::LEDGER_OK) {
     RewardsNotificationService::RewardsNotificationArgs args;
-    AddNotification(RewardsNotificationService::REWARDS_NOTIFICATION_GRANT,
-                    args,
-                    "rewards_notification_grant_" + properties.promotionId,
-                    true);
+    bool should_prevent = rewards_service->GetPreventGrant();
+    if (!should_prevent) {
```

Comment:
Even if FetchGrant is prevented in the panel, a notification will still be added, so we need to be sure to not add it here.